Algorithm.CSharp/FutureOptionBuySellCallIntradayRegressionAlgorithm.cs

```diff
@@ -0,0 +1,137 @@
+    /// portfolio holdings reflect the orders the algorithm has submitted.
+    /// </summary>
+    public class FutureOptionBuySellCallIntradayRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        public override void Initialize()
+        {
+            SetStartDate(2020, 9, 22);
+            typeof(QCAlgorithm)
+                .GetField("_endDate", BindingFlags.NonPublic | BindingFlags.Instance)
+                .SetValue(this, new DateTime(2021, 3, 30));
+
+            // We add AAPL as a temporary workaround for https://github.com/QuantConnect/Lean/issues/4872
+            // which causes delisting events to never be processed, thus leading to options that might never
+            // be exercised until the next data point arrives.
+            AddEquity("AAPL", Resolution.Daily);
+
+            var es18z20 = AddFutureContract(
+                QuantConnect.Symbol.CreateFuture(
+                    Futures.Indices.SP500EMini,
+                    Market.CME,
+                    new DateTime(2020, 12, 18)),
+                Resolution.Minute).Symbol;
+            
+            var es19h21 = AddFutureContract(
+                QuantConnect.Symbol.CreateFuture(
+                    Futures.Indices.SP500EMini,
+                    Market.CME,
+                    new DateTime(2021, 3, 19)),
+                Resolution.Minute).Symbol;
+
+            // Select a future option expiring ITM, and adds it to the algorithm.
+            var esOptions = OptionChainProvider.GetOptionContractList(es19h21, Time)
+                .Concat(OptionChainProvider.GetOptionContractList(es18z20, Time))
+                .Where(x => x.ID.StrikePrice == 3250m && x.ID.OptionRight == OptionRight.Call)
+                .Select(x => AddFutureOptionContract(x, Resolution.Minute).Symbol)
+                .ToList();
+
+            var expectedContracts = new[]
+            {
+                QuantConnect.Symbol.CreateOption(es19h21, Market.CME, OptionStyle.American, OptionRight.Call, 3250m,
+                    new DateTime(2021, 3, 19)),
+                QuantConnect.Symbol.CreateOption(es18z20, Market.CME, OptionStyle.American, OptionRight.Call, 3250m,
+                    new DateTime(2020, 12, 18))
+            };
+
+            foreach (var esOption in esOptions)
+            {
+                if (!expectedContracts.Contains(esOption))
+                {
+                    throw new Exception($"Contract {esOption} was not found in the chain");
+                }
+            }
+
+            Schedule.On(DateRules.Today, TimeRules.AfterMarketOpen(es19h21, 1), () =>
+            {
+                MarketOrder(esOptions[0], 1);
+                MarketOrder(esOptions[1], -1);
+            });
+            Schedule.On(DateRules.Today, TimeRules.Noon, () =>
+            {
+                Liquidate();
+            });
+        }
+
+        public bool CanRunLocally { get; } = true;
```

Comment:
Think we could add an override for the OnEndOfAlgorithm and assert we are no longer invested right? I think the same will apply for all of these regressions algorithms.
Lets please add the missing xml docs too 👍